editoast/src/views/infra/lines.rs

```diff
@@ -0,0 +1,57 @@
+use crate::error::Result;
+use crate::infra::Infra;
+use crate::infra_cache::{InfraCache, ObjectCache};
+use crate::map::BoundingBox;
+
+use crate::DbPool;
+use actix_web::dev::HttpServiceFactory;
+use actix_web::get;
+use actix_web::web::{scope, Data, Json, Path};
+use chashmap::CHashMap;
+use diesel::sql_types::{BigInt, Bool};
+use diesel::QueryDsl;
+use diesel::{ExpressionMethods, RunQueryDsl};
+use serde::Serialize;
+
+/// Return `/infra/<infra_id>/lines` routes
+pub fn routes() -> impl HttpServiceFactory {
+    scope("/lines").service(get_line_bbox)
+}
+
+#[derive(Debug, Serialize)]
+pub struct BboxResponse {
+    geo: Vec<BoundingBox>,
+    sch: Vec<BoundingBox>,
```

Comment:
We don't need this to be a `Vec`. What you want to return is an `InvalidationZone` (it could be renamed).

You have a `union` function to merge the bounding boxes together.